### ISSUE TITLE
mobx @observable conflicts with sort-comp

### ISSUE THREAD

**Issue by bichenkk** [yannickcr/eslint-plugin-react#1194]
In my project using mobx and react, I would like to use the @observable like this followings:
```
{
static methods

@observable variable

life cycle methods
}
```

How do configure the sort-comp rule to allow this situation?

**Comment by yannickcr** [comment]
I'm afraid that `sort-comp` does not handle decorators yet.

**Comment by dgcoffman** [comment]
This worked with eslint-plugin-react@7.4.0

```
        "order": [
          "type-annotations",
          "static-methods",
          "statics",
          "defaultProps",
          "/^observable/",
          "lifecycle",
          "everything-else",
          "render"
        ]
```

but does _not_ work with 7.7.0

**Comment by ljharb** [comment]
@dgcoffman can you clarify what "stopped working" with 7.7, and would you be able to figure out if it worked in 7.5 and 7.6?

**Comment by dgcoffman** [comment]
Sure, 

In 7.4.0, given the config above

``` javascript
class MyComponent extends React.Component<Props> {
  @observable aVariable: string;

  componentWillMount() { ... }
```
is allowed.

In 7.5.0+ (I tried every release), eslint produces `error  aVariable should be placed after componentWillMount  react/sort-comp`

**Comment by ljharb** [comment]
I’m confused; in that example, it’s decorsting “aVariable” - so that’s what should get sorted (the decorator name just goes along for the ride)

**Comment by dgcoffman** [comment]
I don't know the internals of the rule. Maybe up until 7.5.0 is was mis-identifying the variable name as "observable"? We liked it better that way.

**Comment by ljharb** [comment]
Maybe so. Either way tho, the decorator name is irrelevant since it’s not the thing being sorted; we’d have to do something special to allow you to specify that you want the decorator name to apply.

**Comment by TWasilonek** [comment]
Did anyone manage to resolve this?
I have eslint-plugin-react@7.17.0 and still running into this issue.

I tried different combinations mentioned above, and tried to specify the regex a bit more, but nothing works.

```
@observer
class someClass extends Component<Props> {
    @observable someObservable = '';

    static contextType = someContext;

    componentDidMount() { }
}
```
I get this warning:
`someObservable should be placed after componentDidMount eslint(react/sort-comp)`

The same goes for `static contextType`

Current config `.eslintrc.json`:
```
"react/sort-comp": [1, {
      "order": [
        "/^@observable.+$/",
        "static-methods",
        "lifecycle",
        "everything-else",
        "render"
      ]
    }]
```

**Comment by ljharb** [comment]
If this is still an issue with the latest version of the plugin, and someone can comment with repro code, I'll be happy to reopen.
